### PR TITLE
Verify existing snapshot repository

### PR DESCRIPTION
**Pull request by sengineer0**
This PR added a repository verification step when create new snapshot.
also, it add an option to allow delete outdated repo, then create a new one to bypass verification step

Ref: https://github.com/biothings/biothings.api/issues/275

### CODE REVIEW

File: biothings/hub/dataindex/snapshot_repo.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ def create(self, **body):
     def delete(self):
         self.client.snapshot.delete_repository(self.name)
 
+    def verify(self):
+        return self.client.snapshot.verify_repository(self.name)
+
```
Comment (newgene) [inline biothings/hub/dataindex/snapshot_repo.py]: @sengineer0 here using `verify_repository`, it only validates if a repository is working, but we should check if the repo settings match with our config values (e.g. bucket name).

Comment (sengineer0) [reply biothings/hub/dataindex/snapshot_repo.py]: It's updated. Please have a check @newgene

File: biothings/hub/dataindex/snapshot_repo.py [hunk 2]
```diff
@@ -25,6 +25,31 @@ def create(self, **body):
...
+        # Check if the repo pass ElasticSearch's verification
+        self.client.snapshot.verify_repository(self.name)
+
+        # Check if the repo's settings match with the snapshot's config
+        repo_settings = self.client.snapshot.get_repository(self.name)[self.name]
+        incorrect_data = {}
+        
+        config["settings"]["bucket"] = "Test"
+        config["type"] = "test tpe"
```
Comment (newgene) [inline biothings/hub/dataindex/snapshot_repo.py]: @sengineer0 ❓

File: biothings/hub/dataindex/snapshot_repo.py [hunk 3]
```diff
@@ -25,6 +25,31 @@ def create(self, **body):
...
+
+        for field in ["type", "settings"]:
+            if config[field] != repo_settings[field]:
+                incorrect_data[field] = {"config": config[field], "repo": repo_settings[field]}
+        if incorrect_data:
+            raise Exception({
```
Comment (newgene) [inline biothings/hub/dataindex/snapshot_repo.py]: let's use a specific Exception name here.

File: biothings/hub/dataindex/snapshot_repo.py (hunk 2 above)
Comment (sengineer0) [reply biothings/hub/dataindex/snapshot_repo.py]: @newgene  oh, sorry, I am fixing it now :(

File: biothings/hub/dataindex/snapshot_repo.py (hunk 3 above)
Comment (sengineer0) [reply biothings/hub/dataindex/snapshot_repo.py]: ok, sure, will update it soon

Comment (sengineer0) [reply biothings/hub/dataindex/snapshot_repo.py]: It's updated @newgene

File: biothings/hub/dataindex/snapshot_repo.py (hunk 2 above)
Comment (sengineer0) [reply biothings/hub/dataindex/snapshot_repo.py]: it's removed @newgene

**Pull request merged**